### ISSUE TITLE
[DISCUSS] This API should address the Jevons Paradox around increased consumption of verifiable credentials and the rise of their abuses

### ISSUE THREAD

**Issue by kdenhartog** [WICG/identity-credential#35]
> [Jevons Paradox](https://en.wikipedia.org/wiki/Jevons_paradox) is an economic principle which introduces the concept that availability and efficiency of providing a resource (in this case data from a driver's license) can often times lead to increased consumption of the resource. This spec should be addressing whether this additional consumption provides an overall net positive gain for the web platform or if this will cause further harm to marginalized communities of people.
> 
> While many people may consider this a given of any web api, in the specific case of a driver's license we're increasing access to sensitive PII in a verified way about a user. This raises ethical questions which may be in violation of the ethical web principles and privacy principles defined by TAG. Therefore it begs the question to what degree should the web platform be contributing to this phenomena and should this API be standardized in the first place?
> 
> More specifically, I think we need to specifically address what societal impact this additional API will bring to the further marginalization of vulnerable communities and whether the acceptance of this API into the web platform is worth the tradeoffs for those additional harms. By increasing access to sensitive, verified PII we're inadvertently also making it easier to justify the requirement of providing this data in order to access specific web services. While in some cases, many would consider this acceptable such as lawmakers in Louisiana[[1](https://www.npr.org/2023/01/05/1146933317/louisiana-new-porn-law-government-id-restriction-privacy)] it will also introduce a slippery slope where many web services will request this data from a user in a way that stretches the boundary of "legitimate usage" of data. I'm sure we can all think of many more dystopic use cases that could be introduced because of the increased access to this data. For example, I'm sure there will be at least one government that contemplates requiring all users to provide this information to social networks so they can subpoena dissenting voices on social networks like Twitter and surveil or silence them because of this required deanonymization. Is that direction we're prepared to take the web in by introducing this API?

This was originally posted here: https://github.com/WICG/mobile-document-request-api/issues/6 

I don't want this to be lost, so am moving it over here. 

After sitting on this idea a bit I think the direction we should look to technologically here is how we utilize intentional friction in order to reduce the likelihood of abuse. As an example, if we make it so easy to conduct a liveness test with this API then we'll also encourage liveness checks to be done at every authentication check. However, if it has a high level of friction to conduct a liveness check it may be only used for registration and important 2FA step ups.

**Comment by timcappalli** [comment]
@kdenhartog will you be able to join any of the upcoming calls?

**Comment by RByers** [comment]
FWIW using intentional friction to reduce the likelihood of abuse is exactly the direction I've proposed for Chrome's implementation. I imagine some way of computing a risk score (eg. anonymous age verification is lower risk than known unique ID sharing or unknown arbitrary data sharing), then adapting the Chrome/Android UI in response to the risk score. I imagine the details are something we would need to experiment with and iterate on over time, but the principles for enabling such a tradeoff space seem good to be discussing now.

**Comment by kdenhartog** [comment]
> @kdenhartog will you be able to join any of the upcoming calls?

Yup, I intend to. Are the events listed on the WICG calendar or on a separate one? Also, let me make sure I'm a member of the WICG.

**Comment by kdenhartog** [comment]
>  I imagine some way of computing a risk score (eg. anonymous age verification is lower risk than known unique ID sharing or unknown arbitrary data sharing), then adapting the Chrome/Android UI in response to the risk score.

I generally like the direction of this. Most claims could likely be naively calculated (EFF has a good [blogpost](https://www.eff.org/deeplinks/2010/01/primer-information-theory-and-privacy) on this) based on a number of bits of entropy they reveal in which case we could warn the user about this and suggest that too much information is being requested/add additional friction.

**Comment by timcappalli** [comment]
> > @kdenhartog will you be able to join any of the upcoming calls?
> 
> Yup, I intend to. Are the events listed on the WICG calendar or on a separate one? Also, let me make sure I'm a member of the WICG.

Apologies, missed this comment. All meeting details are here: https://github.com/WICG/identity-credential/blob/main/README.md

**Comment by timcappalli** [comment]
@npdoty do you think this is something that PING should discuss vs this group?

**Comment by kdenhartog** [comment]
> do you think this is something that PING should discuss vs this group?

+1 from me. In my opinion, PING and TAG should both be weighing in on these topics early to make sure there's general alignment between the goals of the API and the broader web architecture goals before we start committing to particular solutions.

**Comment by npdoty** [comment]
Definitely a question that PING should discuss, and I've opened a corresponding issue for the w3cping credentials considerations doc.

But also good for the identity-credential group to consider in API design, how we intend to limit overasking or how to introduce friction to prevent that particular trend towards a bad outcome.

**Comment by bvandersloot-mozilla** [comment]
We agree that this is a concern. Excerpted from [Mozilla's standards position](https://github.com/mozilla/standards-positions/issues/1003): 

> A Web that follows [[our principles](https://www.mozilla.org/en-US/about/webvision/full/)](https://www.mozilla.org/en-US/about/webvision/full/) respects an individual’s choice about the identity they present to others. This is an essential part of making the Web a [[safe](https://www.mozilla.org/en-US/about/webvision/full/#safety)](https://www.mozilla.org/en-US/about/webvision/full/#safety) place for [[everyone](https://w3ctag.github.io/ethical-web-principles/#allpeople)](https://w3ctag.github.io/ethical-web-principles/#allpeople). Credentials offered by governments and similar entities generally do not recognize this aspect of identity.
> 
> The potential that requests for digital credentials might  become an expected part of online interactions exists and could damage  this important characteristic of the Web. There are very few online  interactions where providing identity is truly necessary, but many where sites (and governments) might prefer that the identity of a person be  known. Increasing requests for identity does not mean a trivial  annoyance. It could mean that people might be excluded from services and communities when they [[cannot](https://www.unhcr.org/about-unhcr/who-we-protect/stateless-people)](https://www.unhcr.org/about-unhcr/who-we-protect/stateless-people) provide credentials or instead [[choose](https://w3ctag.github.io/privacy-principles/#principle-identity-per-context)](https://w3ctag.github.io/privacy-principles/#principle-identity-per-context) to protect their privacy.

During the last meeting we discussed that it would be useful to decide what use cases are abuse and which aren't, and I think Jevons Paradox is a useful lens to use when building that distinction.

**Comment by DemiMarie** [comment]
I’m opposed to this position, mostly because I can’t see open operating systems (desktop Linux) ever supporting it.

**Comment by johannhof** [comment]
I believe that we'll discuss this in #262, so I'll close out this issue once that's merged. That's not to shut down further discourse on the topic, but to note that we've recorded the concern in the spec and that it might be useful to discuss e.g. potential mitigations in their own issues.

**Comment by timcappalli** [comment]
> What about requiring any site that wishes to use this API to be in an allowlist? Sites would need to request to be added to the allowlist, and would only be added if they legitimately needed the API (most likely for regulatory compliance reasons).

Who maintains the allowlist and who is liable for it? And if you're not on the allowlist, you can still fall back to other methods, which put user data at risk.

Allowlists are a bit of an anti-pattern for the open web.

**Comment by c2bo** [comment]
> > What about requiring any site that wishes to use this API to be in an allowlist? Sites would need to request to be added to the allowlist, and would only be added if they legitimately needed the API (most likely for regulatory compliance reasons).
> 
> Who maintains the allowlist and who is liable for it? And if you're not on the allowlist, you can still fall back to other methods, which put user data at risk.
> 
> Allowlists are a bit of an anti-pattern for the open web.

Yeah, I'd agree that imho the API itself should probably not introduce one single source to govern access to it - ecosystems can and should do so. I think we need to handle this in the individual domains since from everything we've seen so far there are big differences depending on domain/context.

It kinda comes back to the whole scoping discussion: Verifier/Reader Authentication is currently done by the underlying protocols (e.g., OpenID4VP, ISO 18013-7) and not the DC API. DC API governs the invocation, but necessary information and binding to trust ecosystems etc. needs to be provided by the underlying protocols.

**Comment by kdenhartog** [comment]
> Who maintains the allowlist and who is liable for it? And if you're not on the allowlist, you can still fall back to other methods, which put user data at risk.

The DPAs who are regulating the usage of the wallets already too. @burdges explained how this can be used with DPA-CC files above.

In my opinion, this is the proper limitations that should be used to pragmatically balance appropriate usage of the technology with the abuse. It's well understood that these technology needs to be paired with effective policy to limit the abuse which is why trust frameworks are commonly promoted within the space. Don't we want appropriate limitations set in order to limit the abuse of this Web API? If you don't believe these are practical, what counter proposal are you proposing to place these limitations on the API instead?

**Comment by kdenhartog** [comment]
> Yeah, I'd agree that imho the API itself should probably not introduce one single source to govern access to it - ecosystems can and should do so. I think we need to handle this in the individual domains since from everything we've seen so far there are big differences depending on domain/context.
> 
> It kinda comes back to the whole scoping discussion: Verifier/Reader Authentication is currently done by the underlying protocols (e.g., OpenID4VP, ISO 18013-7) and not the DC API. DC API governs the invocation, but necessary information and binding to trust ecosystems etc. needs to be provided by the underlying protocols.


If this is governed at the protocol layer by the verifier controlled domains, how does the client perform enforcement on it to protect the user and act as its agent?

I don't think it's generates a balance of power to say that wallet software needs to be certified by governance bodies and that the data attributes need to be certified by an issuer. Even with issuers, there's an expectation from Verifiers that they're verified by the trust frameworks too. So, it seems reasonable to also set checks and balances on the verifiers ability to rely upon this information correct?

**Comment by DemiMarie** [comment]
Is this API going to be limited to proprietary platforms?  Unless it is feasible to implement this API on open source operating systems I'm opposed to the API altogether.  For instance, any solution that requires remote attestation of the OS is a very bad idea for the same reason that Web Environment Integrity was abandoned.

**Comment by timcappalli** [comment]
> Which wallet implementations exist? Are there ones that are not tied to specific OSs?

There are dozens (if not hundreds) of digital identity wallets available. 

I suggest we take deployment / ecosystem discussions to either a new issue or over to [Digital Credentials Developer Discussions](https://github.com/orgs/digitalcredentialsdev/discussions).

**Comment by kdenhartog** [comment]
@c2bo said:

> Not verifier controlled domains - different trust ecosystems/domains with a verifier and the wallet being part of one or more such trust ecosystems. What entity is the "client" in that case? In my mental model, the main user agent protecting the user should be the wallet for these kind of flows since it likely is best positioned to understand the flows (protocol layer) and specific requirements/policies of the trust ecosystem that it is part of. Establishing trust into Verifier/Issuer will very likely be ecosystem/domain specific and will imho be extremely hard to fully generalize.
> 
> That being said I see value in having the Browser protect the user to a reasonable degree, but that is exactly the layering/scoping discussion we seem to coming back to. There seem to be different mental models on exactly what entity is protecting the user where and how/to what extent.
> 
> The browser fully inspecting presentation requests (i.e., fully parsing & understanding protocols, credential formats, trust ecosystems) is imho problematic at best. It also causes very practical problems in that we would have 4 entities that need to understand all of those parts

The issue here is that we don't believe the OS/App layer is necessary to facilitate this interaction. For example, passkeys today have multiple different methods of storing them such as the Browser, the OS, or in the extension. Currently, I believe you're making the assumption that an identity wallet will only appear as a mobile application. Is this correct?

If, so I'd point to the fact that we currently have a wallet built directly into the browser, proton pass as a password manager extension overrides the credential API to capture the passkey flow first, Browsers do similar things, and it's only when the user intentionally chooses to use a cross device flow (on a desktop) that they're likely using the OS/App flow through.

So, because we don't have guarantees of what level the wallet exists at, we likely need checks in every place still.

**Comment by DemiMarie** [comment]
The question is whether there is a wallet that is not a mobile app and which can store credentials that relying parties will accept.

**Comment by DemiMarie** [comment]
My main concern is that credential **issuers** will only allow issue credentials that they can verify are running on mobile hardware running issuer-trusted OSs.  In such a world, a desktop browser could implement this API, but it wouldn’t be of any use, because it wouldn’t be able to obtain a credential that would actually be trusted.

**Comment by kdenhartog** [comment]
> I have no problem with a browser attempting to fulfill that request natively (which also means parsing the request etc.), but IF a cross-device flow is initiated (which should purely be based upon the user's decision/choice), then it should not be blocked because the browser does not understand the request/protocol.

That's not what I'm arguing for though. My position is that if the browser does not believe the site should be allowed to request this information then it should block the request.

**Comment by kdenhartog** [comment]
> My main concern is that credential issuers will only allow issue credentials that they can verify are running on mobile hardware running issuer-trusted OSs. In such a world, a desktop browser could implement this API, but it wouldn’t be of any use, because it wouldn’t be able to obtain a credential that would actually be trusted.

I generally share this view too that the depending on what assurances are available it will play into this paradox. This point feels akin to whether the browser should recognize a property like authenticator attestations in Webauthn, except in this API it would be about what level of assurance the wallet would contain.

This issue does feel slightly separate from the goal though of restricting how many sites can rely upon this API. @DemiMarie would you mind if we move this to a separate issue so that we can discuss this further?

**Comment by DemiMarie** [comment]
@kdenhartog done: #349

**Comment by marcoscaceres** [comment]
This issue raised concerns about the Jevons Paradox — that increased availability of digital credentials could accelerate abusive credential demands.

The spec now addresses this directly in the **"Mitigating unnecessary requests for government credentials"** section:

> "A major concern with increased availability of government credentials online is Jevon's Paradox, i.e., the chance of increasing demand for credentials through lower friction of access."

The spec further covers the specific risks raised in this issue: prompt fatigue, surveillance, exclusion/discrimination, and non-government credential proliferation. It recommends higher friction for exchanges without verifier authentication and discusses the need for broader policy frameworks.

We believe this issue is now addressed by the current spec text. @kdenhartog, could you confirm within two weeks (by May 14) whether you're satisfied this covers your original concern? If we don't hear back, we'll proceed to close this issue.

**Comment by kdenhartog** [comment]
Am I understanding correctly the proposed solution here is to acknowledge the problem in the spec, without normatively defining the [proposal included](https://github.com/w3c-fedid/digital-credentials/issues/35#issuecomment-2315300567) in this thread?

If that's the case, as I [stated previously](https://github.com/w3c-fedid/digital-credentials/issues/35#issuecomment-3776574295) I wouldn't consider this a solution to the issue.

**Comment by marcoscaceres** [comment]
OK, as the algorithms are all there now, can you send us a suggestion for how to update the spec?

**Comment by marcoscaceres** [comment]
I’ll take your 👍 as an offer to help us out. To set context for anyone reading this thread later, here's a summary of how the current spec (WD 19 April 2026) addresses the Jevons concern.

**Friction baked into every call (normative):**
- Express user permission is required; mediation is implicitly and non-overridably `"required"`
- Transient activation is required, so sites can't query in the background
- User activation is consumed on each call, so each request is a discrete user action
- Requests must be unencrypted, enabling user-agent inspection for risk analysis
- Two separate Permissions Policy features (`digital-credentials-get` and `digital-credentials-create`), each defaulting to `'self'`, so embedded contexts only get the capability they need
- Only Secure Contexts can use the API
- The Credential Request Coordinator enforces at most one active interaction per top-level traversable

**Closed set of supported protocols (§5):**
- The spec hardcodes the allowed presentation and issuance protocols via WebIDL enums (`openid4vp-v1-*`, `org-iso-mdoc`, `openid4vci-v1`); requests for any other protocol are dropped during validation
- Each protocol included was vetted by the WG; new protocols require WG consensus to be added rather than self-registration
- This is a meaningful tightening compared to the earlier open registry approach: every protocol now has to clear a higher bar (WG consensus, baked directly into IDL) rather than passing through a registry inclusion process
- The protocols included in §5 support response encryption, selective disclosure, mechanisms to reduce "phone home" reliance, verifier authorization where applicable, and user transparency/consent surfaces (these properties are discussed as expectations on protocols in §11.3.1)

**Documented in privacy considerations (non-normative but substantive):**
- The Jevons effect is named explicitly in §11.4.1.2 ("Risk of proliferation of requests for government credentials"), with the four concrete harms enumerated: information leakage, prompt fatigue, surveillance/pseudonymity loss, and exclusion/discrimination
- Higher friction and clearer messaging is recommended for government-credential exchanges without verifier authentication
- User agents are advised to apply increased friction for unrecognized credential types
- An interoperable abuse-reporting system for verifiers is flagged as something to consider
- Avoiding leaks of credential availability is addressed in §11.5.4. Sites cannot learn whether a credential is available without going through the user permission flow

So the spec isn't treating proliferation as someone else's problem: it's shaped a number of design choices around exactly this risk, including the recent decision to drop the open registry in favor of a closed protocol set. 

@kdenhartog, the open question your proposal will help us answer is whether additional normative requirements are warranted on top of this baseline, and where they should land.

**Comment by kdenhartog** [comment]
Yes it is. I'll get a pull request up for a more specific proposal of normative changes to [section 6.3](https://www.w3.org/TR/digital-credentials/#validate-credential-requests) by May 18th for this. I know it's IIW week, so figure most of this is coming from doing in person issue triage.

The underlying concern here remains that this is a generic API to pass 3P attested personal data around in various credential formats and we're still architecturally utilizing a deny list approach to reduce this. The proposal put forth previously is moves to an allow list approach instead. Considering we have many laws coming in globally which are requiring the usage of these tools, but not providing legal enforcement to prevent abuse of the tools it seems like it's falling on the APIs to prevent this abuse. As one example, EUDI had this allow list approach in place and [removed it](https://epicenter.works/en/content/eu-commission-undermines-eidas-protections-again).

We'll go from there in terms of next steps on this.

**Comment by c2bo** [comment]
> Yes it is. I'll get a pull request up for a more specific proposal of normative changes to [section 6.3](https://www.w3.org/TR/digital-credentials/#validate-credential-requests) by May 18th for this. I know it's IIW week, so figure most of this is coming from doing in person issue triage.
> 
> The underlying concern here remains that this is a generic API to pass 3P attested personal data around in various credential formats and we're still architecturally utilizing a deny list approach to reduce this. The proposal put forth previously is moves to an allow list approach instead. Considering we have many laws coming in globally which are requiring the usage of these tools, but not providing legal enforcement to prevent abuse of the tools it seems like it's falling on the APIs to prevent this abuse. As one example, EUDI had this allow list approach in place and [removed it](https://epicenter.works/en/content/eu-commission-undermines-eidas-protections-again).
> 
> We'll go from there in terms of next steps on this.

EU never followed a full-blown allow-list - the EU followed (and still does) a model where a Relying Party has to be identified and has to register its use-case (what data they are asking for and why). That was never removed, but there are ongoing discussions how exactly that information is conveyed to the wallet (online lookup vs RP registration certificate) - that is also what epicenter referenced (which I agree with btw).

I believe we need something like this - which allows users to properly report abuse etc. The big question is where exactly that layer lives. The way that system is currently being defined in Europe is via extension points / protocol extensions in the underlying protocols, so it would be outside of direct control of DC API.

A very concrete example for such a mechanism would be the extension points that were built for exactly that purpose in the underlying protocols, such as [Verifier Info](https://openid.github.io/OpenID4VP/openid-4-verifiable-presentations-1_0-wg-draft.html#name-verifier-info) for OpenID4VP. That is the extension point that is for example planned to be used to inform the user about the registered use-case, purpose, data protection agency contact information etc. for EUDI wallets.

**Comment by hlflanagan** [comment]
See [4 May 2026 meeting notes](https://github.com/w3c-fedid/meetings/blob/main/2026/2026-05-04-DCAPI-A-notes.md#aob).

**Comment by kdenhartog** [comment]
The proposal I'm putting forward here is that the browser is the enforcement point for those checks not the wallet and if they aren't met then the browser will error the request.

I don't think people are understanding my concern here. We (Brave) see it as a feature for the API to only be allowed on a limited number of sites which have undergone additional scrutiny and for the browser to be the enforcement layer for this as it's the first stop in the life cycle of a request for personal data. Hence it should be the enforcement layer for the API, not a pipe to the wallet which leaks information to sites about apps that exist on the users device. That then presents new fingerprinting concerns as well.

That's the compromise I'm willing to make here to address our original objection. If we're unable to find a compromise on this though then we're forced to re-object here at the candidate recommendation stage which is what I'm trying to avoid.

> A very concrete example for such a mechanism would be the extension points that were built for exactly that purpose in the underlying protocols, such as [Verifier Info](https://openid.github.io/OpenID4VP/openid-4-verifiable-presentations-1_0-wg-draft.html#name-verifier-info) for OpenID4VP. That is the extension point that is for example planned to be used to inform the user about the registered use-case, purpose, data protection agency contact information etc. for EUDI wallets.

This is making an assumption on a particular protocol which sets a requirement on every other protocol to then support it as well. Otherwise, browsers have to implement protocol specific logic for each new one they support in order to protect their users.

**Comment by timcappalli** [comment]
> The proposal I'm putting forward here is that the browser is the enforcement point for those checks not the wallet

> We (Brave) see it as a feature for the API to only be allowed on a limited number of sites which have undergone additional scrutiny and for the browser to be the enforcement layer for this as it's the first stop in the life cycle of a request for personal data

These statements, and your proposed compromise are fundamentally incompatible with the verifiable digital credentials ecosystem.

> not a pipe to the wallet which leaks information to sites about apps that exist on the users device. 

Users are in full control of whether a request gets from the browser to a credential manager. They're also in full control of releasing a presentation.

**Comment by marcoscaceres** [comment]
@kdenhartog let’s get that pull request up. 🙏 That will hopefully give us something concrete to work with. 

I’d like to close this issue out in two weeks with a PR, but let us know if you need more time.

**Comment by marcoscaceres** [comment]
@kdenhartog gentle ping on a PR. Otherwise please let us know if the above mitigations are sufficient to closed this out.

**Comment by marcoscaceres** [comment]
Just FYI, we are planning on closing this on 18th to stay on track to meet our October deadline for CR. We can, of course, reopen once we receive additional information or a PR. 

@kdenhartog, if you'd like to have a call about this, I'm happy to go over anything with you and assist with a PR if that would help.

**Comment by DemiMarie** [comment]
I don’t think anything less than an allowlist maintained by a nonprofit organization would be suitable.

Access to this API should be limited to sites that can justify, to the community, that they _must_ use it to comply with a specific, applicable law or regulation in a jurisdiction they operate in.  It’s a compliance tool, nothing more.

**Comment by marcoscaceres** [comment]
@DemiMarie, I don't disagree. But that's way outside the scope of the working group. So, the question is (at least in my mind):

* Does the API or the two formats prevent such an oversight organization from being established? 
* Do the formats allow arbitrary sites from requesting credentials without first going through such an authority? 

It's a laudable goal, and I guess the spec could suggest the establishment of such an authority? A kinda precedent I can think of for this, though not exactly the same (you need to squint real hard), is [Let's Encrypt](https://letsencrypt.org), which vastly helped transition the web to TLS by default.

**Comment by DemiMarie** [comment]
@marcoscaceres I think something like the following might work, **if** browsers can be trusted to enforce it:

1. Unless the domain is localhost or on public allowlist, the request unconditionally fails.  An organization (EFF?  ISRG?) should maintain a list of domains that are allowed to use the API.
2. The process of adding domains should be public.  Domains will only be added if they can justify, to the community, that they must use it to comply with a specific, applicable law or regulation in a jurisdiction they operate in.
3. Domains must justify each piece of information they want, and why they can’t use a zero-knowledge proof instead.
4. The only supported verifiers are verifiers located on hardware external to the device, such as a physical smart card or security token, that can be used with any computer via a standardized protocol.

- Clause 1 ensures that domains cannot use the API unless allowed to do so, and that the list of domains that may use it is public.
- Clause 2 specifies that use of the API will only be allowed when the alternative would be to not operate.
- Clause 3 requires that the information revealed be kept to a minimum.
- Clause 4 prevents using the API to discriminate against specific types of computing hardware or software.  In particular, it means that implementations tied to the secure element in a mobile device are forbidden.  Instead, the verifier must be in a _separate_ piece of hardware that can be carried between machines and used by any computer.

@vivianlazaras do you have any thoughts on this?

**Comment by marcoscaceres** [comment]
@DemiMarie, sorry to push you a little bit on this, but you kinda side stepped my two questions. I think those are fundamental and would like your answers on those, as I think they really matter. Asking them again:

* Does the API or the two formats prevent such an oversight organization from being established? 
* Do the formats allow arbitrary sites from requesting credentials without first going through such an authority? 

(these are not *gotcha* questions. I'm asking them sincerely as I don't fully know the answers) 

The comparison to ISRG's Let's Encrypt is apt, in that a user agent could potentially do what you proposed. I think this relates to https://github.com/w3c-fedid/digital-credentials/issues/472 (the proposed prose there would more or less cover what you are proposing)... but also, it would be premature to mandate that the user agent use such an authority without that authority first being established. As with Let's Encrypt, that came after TLS to solve a particular problem. 

I'll also note that even relying on ISRG is a form of centralization (the "who watches the watcher" problem and single point of failure). There's definitely a role for an authority like that to play, but limiting to one seems risky no matter how well intentioned.

**Comment by DemiMarie** [comment]
@marcoscaceres It’s hard to tell offhand because the spec is complicated.  That said, the browser is always allowed to simply refuse to perform the operation, so the answers are “no” and “maybe, depending on applicable law and who issues the credentials”.

That said, I don’t think after-the-fact is an option in this case, becaues it will break too many sites.  That’s why I’m asking for an up-front approach, where the authority is established first.  It ensures that _every_ site that uses the API has gone through checks.

**Comment by marcoscaceres** [comment]
Right, but the API has already shipped and it’s actively being used, so we are already living in the “after the fact” world. However, nothing prevents what you are proposing from being established in the future (again citing Let’s Encrypt as a fine example that made a huge difference establishing after the fact). 

I’ll like to propose we close this issue as I think all points have been made and the working group has made a concerted effort to address the OP’s concerns directly in the spec (and actively continues to add friction, like, for example #519 and #520). To your point, “the spec is complicated” - and that’s by design. I encourage you to attempt to create a request with either OpenID or ISO18013… it’s, as we say in Australia, bloody hard and a ton of work. 

Closing this also doesn’t mean we can’t reopen the issue when there’s new information and we’d still gladly take a pull request as requested above. We’d also invite anyone here to jump on one of our WG calls for a high bandwidth discussion.

**Comment by marcoscaceres** [comment]
Closing as I think we've adequately addressed this in the spec now. 

As discussed above, there's a number of things that might assist with adding friction to reduce the impact of  Javons paradox (e.g., requiring response encryption #519).
